src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Collection/IDictionaryConverter.cs

```diff
@@ -27,33 +28,12 @@ protected override void Add(string key, in object? value, JsonSerializerOptions
 
         protected override void CreateCollection(ref Utf8JsonReader reader, ref ReadStack state)
         {
-            JsonTypeInfo typeInfo = state.Current.JsonTypeInfo;
-
-            if (TypeToConvert.IsInterface || TypeToConvert.IsAbstract)
+            base.CreateCollection(ref reader, ref state);
+            TDictionary returnValue = (TDictionary)state.Current.ReturnValue!;
+            if (returnValue.IsReadOnly)
```

Comment:
should this logic be moved to base?

Reply:
`IsReadOnly` is a different property depending on whether the converted type is `IList<T>`, `ICollection<T>`, `IList` or `ICollection` so we can't really extract it to a common base.